main.py

```diff
@@ -137,7 +144,16 @@ def bindDropdownWidget(widget, var):
 		bindCheckWidget(self.waitForGdb, 'gdb_wait')
 		bindTextWidget(self.gdbPort, 'gdb_port')
 		bindTextWidget(self.additionalArgs, 'extra_args')
+		
+		# Controller UI has additional logic
+		self.controller1.currentIndexChanged.connect(updateControllerUi)
+		self.controller2.currentIndexChanged.connect(updateControllerUi)
+		self.controller3.currentIndexChanged.connect(updateControllerUi)
+		self.controller4.currentIndexChanged.connect(updateControllerUi)
```

Comment:
nit: this could be be more optimized / better designed (where this happens only for the required widget). But this is fine for now.

Reply:
Agree